packages/widget/src/stores/form/FormUpdater.tsx

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { useEffect, useRef } from 'react';
+import type { FormFieldNames } from './types';
+import { useAccount } from '../../hooks';
+import { useFieldActions } from './useFieldActions';
+import { isItemAllowed, useWidgetConfig } from '../../providers';
+import type { DefaultValues } from './types';
+import { FormKey } from './types';
+
+export const FormUpdater: React.FC<{
+  defaultValues: Partial<DefaultValues>;
+}> = ({ defaultValues }) => {
+  const { fromChain, toChain, chains } = useWidgetConfig();
+  const { account } = useAccount();
+  const { isTouched, resetField, setFieldValue, getFieldValues } =
+    useFieldActions();
+  const previousDefaultValues = useRef(defaultValues);
+
+  // Set wallet chain as default if no chains are provided by config and if they were not changed during widget usage
+  useEffect(() => {
+    const chainAllowed =
+      account.chainId && isItemAllowed(account.chainId, chains);
+    if (!account.isConnected || !account.chainId || !chainAllowed) {
+      return;
+    }
+
+    if (!fromChain && !isTouched('fromChain') && !isTouched('fromToken')) {
+      resetField('fromChain', { defaultValue: account.chainId });
+      setFieldValue('fromToken', '');
+      if (isTouched('fromAmount')) {
+        setFieldValue('fromAmount', '');
+      }
+    }
+    if (!toChain && !isTouched('toChain') && !isTouched('toToken')) {
+      resetField('toChain', { defaultValue: account.chainId });
+      setFieldValue('toToken', '');
+    }
+  }, [
+    account.chainId,
+    account.isConnected,
+    chains,
+    fromChain,
+    toChain,
+    isTouched,
+    resetField,
+    setFieldValue,
+  ]);
+
+  // Makes widget config options reactive to changes
+  // Acts similar to values property from useForm, but includes additional logic for chains
```

Comment:
Should be updated I guess, we no longer have `useForm`

Reply:
Will update